packages/block-editor/src/components/link-control/index.js

```diff
@@ -219,20 +253,52 @@ function LinkControl( {
 			const isInternal = startsWith( val, '#' );
 			const isTel = protocol.includes( 'tel' );
 
-			const handleManualEntry =
+			const maybeURL =
 				isInternal ||
 				isMailto ||
 				isTel ||
 				isURL( val ) ||
 				( val && val.includes( 'www.' ) );
 
-			return handleManualEntry
+			return maybeURL
 				? handleDirectEntry( val, args )
 				: handleEntitySearch( val, args );
 		},
 		[ handleDirectEntry, fetchSearchSuggestions ]
 	);
 
+	const handleOnCreate = async ( entityTitle ) => {
+		let newEntity;
+
+		setIsResolvingLink( true );
+		setErrorMsg( null );
+		try {
+			newEntity = await createEntity( 'page', entityTitle );
```

Comment:
@aduth It feels like the `page` should not be baked into this component. Rather the consuming component (in this case Nav Link Block) should provide the "type". LinkControl should be unaware of the type of entity it is creating. Do you agree?

This was refactored by @marekhrabe so I could have things wrong.